src/App.tsx

```diff
@@ -1,20 +1,55 @@
 import { SearchHeader } from "./components/header/SearchHeader";
-import { ClothingCardsGrid } from "./components/clothingGrid/ClothingCardsGrid";
+import { ProductsGrid } from "./components/productsGrid/ProductsGrid";
 import { Sidebar } from "./components/sidebar/Sidebar";
 import { useState } from "react";
+import { useDebounce } from "./hooks/useDebounce";
 
 function App() {
   const [isSidebarOpen, setIsSidebarOpen] = useState(false);
+  const [searchTerm, setSearchTerm] = useState("");
+  const [sortValues, setSortValues] = useState({
+    sortBy: "",
+    sortOrder: "",
+  });
+  const { sortBy, sortOrder } = sortValues;
+  const debouncedSearchTerm = useDebounce(searchTerm, 500);
```

Comment:
Nice job, creating the custom hook. You can check `@uidotdev/usehooks` that already solves it
https://usehooks.com/usedebounce